lib/exceptional_synchrony/callback_exceptions.rb

```diff
@@ -21,13 +21,13 @@ def map_deferred_result(deferrable)
             result
           end
         when :failed
-          if result.respond_to?(:error) && result.error =~ /timeout/i
-            raise Timeout::Error
+          if result.respond_to?(:error)
+             handle_result_error(result)
           else
-            raise Failure, result.inspect
+            raise Failure, format_failure(nil.inspect, format_inspect(result))
```

Comment:
By omitting this optional parameter here, the method can do a better job than saying ERROR = nil. Maybe it would omit the ERROR = part entirely?